File: editorial/chap06/cpp/answer_B40.cpp
#include <iostream>
using namespace std;
 
long long N, A[200009];
long long cnt[109];
long long Answer = 0;
 
int main() {
	// 입력
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> A[i];
 
	// 개수를 센다
	for (int i = 0; i < 100; i++) cnt[i] = 0;
	for (int i = 1; i <= N; i++) cnt[A[i] % 100] += 1;
 
	// 답을 구한다
	for (int i = 1; i < 50; i++) Answer += cnt[i] * cnt[100 - i];
	Answer += cnt[0] * (cnt[0] - 1LL) / 2LL;
	Answer += cnt[50] * (cnt[50] - 1LL) / 2LL;
 
	// 출력
	cout << Answer << endl;
	return 0;
}

File: editorial/chap06/python/answer_B40.py
# 입력
N = int(input())
A = list(map(int, input().split()))
 
# 개수를 센다
cnt = [ 0 ] * 100
for i in range(N):
	cnt[A[i] % 100] += 1
 
# 답을 구한다
Answer = 0
for i in range(1, 50):
	Answer += cnt[i] * cnt[100-i]
Answer += cnt[0] * (cnt[0] - 1) // 2
Answer += cnt[50] * (cnt[50] - 1) // 2
 
# 출력
print(Answer)
